src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/InvokeRestMethodCommand.Common.cs

```diff
@@ -91,45 +91,34 @@ internal override void ProcessResponse(HttpResponseMessage response)
                 }
                 else
                 {
-                    // Determine the response type
-                    RestReturnType returnType = CheckReturnType(response);
-
                     // Try to get the response encoding from the ContentType header.
                     string charSet = WebResponseHelper.GetCharacterSet(response);
 
                     string str = StreamHelper.DecodeStream(responseStream, charSet, out Encoding encoding, _cancelToken.Token);
 
-                    object obj = null;
-                    Exception ex = null;
-
-                    string encodingVerboseName;
-                    try
-                    {
-                        encodingVerboseName = string.IsNullOrEmpty(encoding.HeaderName) ? encoding.EncodingName : encoding.HeaderName;
-                    }
-                    catch (NotSupportedException)
-                    {
-                        encodingVerboseName = encoding.EncodingName;
-                    }
-
                     // NOTE: Tests use this verbose output to verify the encoding.
-                    WriteVerbose(string.Create(System.Globalization.CultureInfo.InvariantCulture, $"Content encoding: {encodingVerboseName}"));
+                    WriteVerbose(string.Create(System.Globalization.CultureInfo.InvariantCulture, $"Content encoding: {encoding.HeaderName}"));
```

Comment:
HeaderName can throw.

Reply:
I don't think it will throw
Test:
```ps1
[System.Text.Encoding]::GetEncodings().GetEncoding().Count -eq `
([System.Text.Encoding]::GetEncodings().GetEncoding() | % {$_.EncodingName}).Count -eq ` ([System.Text.Encoding]::GetEncodings().GetEncoding() | % {$_.HeaderName}).Count -eq 116 #tested on windows
```

Reply:
There is `throw` in implementation.

Reply:
EncodingName: https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Text/Encoding.cs,355
HeaderName: https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Text/Encoding.cs,371
Both can throw in the same condition: https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Text/Encoding.cs,327

Reply:
I don't understand - you point NotSupportedException in the implementation but says the try-catch is not correct.

Reply:
3 is good catch and we could use String.Empty.
I cannot agree with 1. and 2. points since it is not safe as the previous code can be changed and also there is dynamic code pages registration.